model/task_lifecycle.go

```diff
@@ -1368,9 +1377,13 @@ func MarkTasksReset(taskIds []string) error {
 		return errors.WithStack(err)
 	}
 
-	if err = task.ResetTasks(tasks); err != nil {
+	info, err := task.ResetTasks(tasks)
+	if err != nil {
 		return errors.Wrap(err, "resetting tasks in database")
 	}
+	if info.Updated == 0 {
```

Comment:
Same qusetion as above about idempotency of operations like marking a task reset. If we exit early here, we might be unable to retry updating dependencies.

Reply:
Understood, following same pattern as resetting a single task now.